example_test.go

```diff
@@ -42,11 +43,13 @@ func ExampleNfqueue_Register() {
 	}
 
 	// Register your function to listen on nflqueue queue 100
-	err = nf.Register(ctx, fn)
-	if err != nil {
-		fmt.Println(err)
-		return
-	}
+	err = nf.RegisterWithErrorFunc(ctx, fn, func(e error) int {
+		if err != nil {
+			fmt.Println(err)
+			return -1
+		}
+		return 0
+	})
```

Comment:
If [ErrorFunc](https://pkg.go.dev/github.com/florianl/go-nfqueue#ErrorFunc) is called in the scope of `RegisterWithErrorFunc()`, then error is never nil.
But `RegisterWithErrorFunc()` also returns an error, e.g. if the hook can not be registered to the socket. Then this error should not be ignored.
```suggestion
	err = nf.RegisterWithErrorFunc(ctx, fn, func(e error) int {
		fmt.Println(err)
		return -1
	})
	if err != nil {
		fmt.Println(err)
		return
	}
```

Reply:
Sorry, I misunderstood your original request. Changed as suggested.

Reply:
Thanks for the update - it was my mistake not to be precious about my requested changes.